bam_to_mate_hist.py

```diff
@@ -149,19 +164,71 @@ def calc_n50_from_header(header, xx=50.0):
     cumsum = 0
     contig_len = 0
     for length in reversed(lens):
-        contig_len = length
+        contig_len = int(length)
         cumsum += length
         if cumsum >= nxx_len:
             break
-    return contig_len, total
+    greater_10k = [contig for contig in lens if contig > 1e4]
+    return contig_len, total, len(greater_10k)
+
+
+def html_from_judgement(good, bad):
+    '''return a formatted HTML string based on two judgment bool values
+    '''
+
+    if good and not bad:
+        return '<span style="background-color:green">PASS</span>'
+    elif not good and bad:
+        return '<span style="background-color:red">FAIL</span>'
+    elif good and bad:
+        return '<span style="background-color:yellow">MIXED RESULTS</span>'
+    elif not good and not bad:
+        return '<span style="background-color:yellow">LOW SIGNAL</span>'
+    else:
+        raise ValueError("logical impossibility!")
+
+
+def hic_library_judger(out_dict):
+    '''Pass judgement on the library according to certain mostly subjective ideas about what is good
+
+    Args:
+        stat_dict ({str: float/str}): mapping of lib characteristics to their values
+
+    Returns:
+        judgement (str): an HTML string to put into pass/fail box
+    '''
+    long_contacts = (float(out_dict["NUM_10KB_PAIRS"]) > 0.05)
+    long_floor = (float(out_dict["NUM_10KB_PAIRS"]) > 0.01)
+    useful_contacts = (float(out_dict["NUM_DIFF_CONTIG_PAIRS"]) > 0.3)
+    low_contiguity = (out_dict["N50"] < 100000)
+    many_zero_pairs = (float(out_dict["ZERO_DIST_PAIRS"]) > 0.1)
+    many_many_zero_pairs = (float(out_dict["ZERO_DIST_PAIRS"]) > 0.2)
+    high_dupe = (float(out_dict["NUM_DUPE_READS"]) > 0.1 and out_dict["NUM_READS"] <= 1e6) \
+                or (float(out_dict["NUM_DUPE_READS"]) > 0.3 and out_dict["NUM_READS"] <= 1e9)
+
+    print long_contacts, long_floor, useful_contacts, low_contiguity, many_zero_pairs, many_many_zero_pairs, high_dupe
+    if (long_contacts or useful_contacts) and (low_contiguity or long_floor):
+        good = True
+    else:
+        good = False
+
+    bad = False
+    if low_contiguity:
+        if many_many_zero_pairs or high_dupe:
+            bad = True
+    else:
+        if many_zero_pairs or high_dupe or not long_floor:
+            bad = True
+    return html_from_judgement(good, bad)
 
 
 def make_histograms(dists, bamfile, outfile_name):
     '''make the read distance histograms using matplotlib and write them to disk.
-	Args:
-		dists (numpy array of ints): Distances to plot in histogram.
-		bamfile (str): path to bamfile of dists
-	'''
+    Args:
+        dists (numpy array of ints): Distances to plot in histogram.
+        bamfile (str): path to bamfile of dists
+    '''
+
     dists = dists[[dist > 0 for dist in dists]]
```

Comment:
This looks like it will make dists equal dists[num_dists_gt_0].  Is this the intended behavior? Unsure what the structure of dists is supposed to be to understand why you want to access dists[num_dists_gt_0], but if this is correct then disregard.

Reply:
yes. some elements are not relevant to the distance calculations but are merely placeholders, and these have the value -1. we want to exclude those because they are not valid distances.